services/configServices/NetlifyTomlService.js

```diff
@@ -0,0 +1,39 @@
+const axios = require("axios")
+const toml = require("toml")
+
+// Import error types
+const { NotFoundError } = require("@errors/NotFoundError")
+
+const validateStatus = require("@utils/axios-utils")
+
+const { GITHUB_BUILD_ORG_NAME, GITHUB_BUILD_REPO_NAME } = process.env
+
+class NetlifyTomlService {
+  async read({ accessToken }) {
+    const endpoint = `https://api.github.com/repos/${GITHUB_BUILD_ORG_NAME}/${GITHUB_BUILD_REPO_NAME}/contents/overrides/netlify.toml`
+
+    const resp = await axios.get(endpoint, {
+      validateStatus,
+      headers: {
+        Authorization: `token ${accessToken}`,
+        "Content-Type": "application/json",
+      },
+    })
+
+    if (resp.status === 404)
+      throw new NotFoundError("netlify.toml file does not exist")
+
+    const { content } = resp.data
+
+    // Convert to readable form
+    const netlifyTomlReadableContent = toml.parse(Base64.decode(content))
+
+    // Headers is an array of objects, specifying a set of access rules for each specified path
+    // Under our current assumption, we apply the first set of access rules to all paths
```

Comment:
i think i might be lacking context - could i clarify what's the current assumption here?

Reply:
Our assumption is that the file only contains a single set of access rules, so we take the first set!

Reply:
could we write that down in the comment too?

Reply:
Sure, done in https://github.com/isomerpages/isomercms-backend/pull/357/commits/6517e43054b62063d128deefc695551a26b78157